hs_core/views/__init__.py

```diff
@@ -344,18 +387,53 @@ class AddGroupForm(forms.Form):
         frm = AddGroupForm(data=request.POST)
         if frm.is_valid():
             res.edit_groups.add(frm.cleaned_data['group'])
-    elif t == 'add_owner':
-        frm = AddUserForm(data=request.POST)
-        if frm.is_valid():
-            res.owners.add(frm.cleaned_data['user'])
-    elif t == 'make_public':
-        #if res.metadata.has_all_required_elements():
-        if res.can_be_public:
-            res.public = True
-            res.save()
-    elif t == 'make_private':
-        res.public = False
-        res.save()
+    else:
+        # change permissions in HSAccess Database for hydroshare access control
+        try:
+            ha_obj = HSAlib.HSAccess(str(request.user.username), 'unused', django_settings.HS_ACCESS_DB, django_settings.HS_ACCESS_USERNAME,
+                                     django_settings.HS_ACCESS_PASSWORD, django_settings.HS_ACCESS_HOST, django_settings.HS_ACCESS_PORT)
+        except:
+            # unable to connect to the database
+            raise DatabaseError("unable to connect to the database HSAccess.")
+
+        try:
+            if t == 'add_view_user':
+                frm = AddUserForm(data=request.POST)
+                if frm.is_valid():
+                    res.view_users.add(frm.cleaned_data['user'])
+                    new_user = frm.cleaned_data['user']
+                    new_uuid = ha_obj.get_user_uuid_from_login(str(new_user.username))
```

Comment:
maybe the form should be the thing responsible for these actions, the same way a form is responsible for saving a model or logging in a user.